demux/utils/samplesheet.py

```diff
@@ -258,13 +262,12 @@ def _validate_unique_index(samplesheet):
                         if index not in sample_of:
                             sample_of[index] = set()
                         sample_of[index].add(line["sample_id"])
+
                     for index, samples in sample_of.items():
                         if len(samples) > 1:
-                            samples_list = list(samples)
-                            samples_list.sort()
```

Comment:
This was in my PR and need to be in place. You removed stuff that was supposed to be in there!